Show error message in `@now/go` when using `package main`

Its not clear to new users that `package main` cannot be used with `go.mod`.

The current error message needs improvement:

```
./main__now__go__.go:8:6: main redeclared in this block
previous declaration at ./main.go:36:6
```

Instead, we could use the AST to check if `package main` is being used and show a better error such as "Please change `package main` to `package handler`".

Another alternative is to rewrite the file for the user to avoid this problem altogether.

--- a/packages/now-go/index.ts
+++ b/packages/now-go/index.ts
@@ -163,6 +163,11 @@ Learn more: https://zeit.co/docs/v2/deployments/official-builders/go-now-go/#ent
   // using `go.mod` way building the handler
   const packageName = parsedAnalyzed.packageName;
   const isGoModExist = await pathExists(join(entrypointDirname, 'go.mod'));
+
+  if (isGoModExist && packageName === 'main') {
+    throw new Error('Please change `package main` to `package handler`');
+  }
+
   if (packageName !== 'main') {
     const go = await createGo(
       goPath,

